pkg/services/ngalert/api/authorization.go

```diff
@@ -136,13 +150,15 @@ func (api *API) authorize(method, path string) web.Handler {
 	case http.MethodDelete + "/api/alertmanager/grafana/config/api/v1/alerts": // reset alertmanager config to the default
 		eval = ac.EvalPermission(ac.ActionAlertingNotificationsDelete)
 	case http.MethodGet + "/api/alertmanager/grafana/config/api/v1/alerts":
+		fallback = ac.ReqOrgAdminOrEditor // moved from the request handler as is
 		eval = ac.EvalPermission(ac.ActionAlertingNotificationsRead)
```

Comment:
This is somewhat confusing.

My understanding is that `eval` would only contain the permission if FGAC is enabled. However, you're assigning two variables when you know ahead of time that only of them might not return something. 

While somewhat innocuous is not very obvious and can lead to harder debugging. How do you feel about something like:

```golang
type scopedAction struct {
	action string
	scopes []string
}

func evalWithFallback(actions []scopedAction, fallback string) Evaluator {
	if len(actions) == 1 {
		// evaluate one
	}

	// evaluateAny

	return fallback
}
```

With that, you could replace this to:

```golang
evalWithFallback(ac.ActionAlertingNotificationsRead, ac.ReqOrgAdminOrEditor)
```

Reply:
exactly this happens several line below 
https://github.com/grafana/grafana/pull/46931/files#diff-627a793303894bbf1b53347d4c9c0316e28159fa392b13786f80553dc71e8dd4R185

This method called only when Grafana server starts and registers API routes. Once per route.

Reply:
Discussed offline, I misunderstood what was going on here.